### PR TITLE
Support Multiple Fee Payers

### PR DESCRIPTION
**Pull request by sddioulde**
### Description

Fetch random solana fee payer from identity and send it through solana relay for rewards and user bank creation.

Here is the protocol counterpart PR: https://github.com/AudiusProject/audius-protocol/pull/2453

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?

### How Has This Been Tested?

Locally by spinning up all services and the client and making sure we get the random fee payer and that it's passed in the right places and that those functionalities still work.

Also updated unit test.

### How will this change be monitored?

For features that are critical or could fail silently please describe the monitoring/alerting being added.

### CODE REVIEW

File: src/store/solana/sagas.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { put, call, take, select } from 'typed-redux-saga'
+
+import { fetchAccountSucceeded } from 'common/store/account/reducer'
+import { getFeePayer } from 'common/store/solana/selectors'
+import { setFeePayer } from 'common/store/solana/slice'
+import AudiusBackend from 'services/AudiusBackend'
+
+function* watchForFeePayer() {
+  console.log('watching for fee payer')
```
Comment (sddioulde) [inline src/store/solana/sagas.ts:9]: going to remove

File: src/store/solana/sagas.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  yield take(fetchAccountSucceeded.type)
+  const feePayer = yield* call(AudiusBackend.getRandomFeePayer)
+  const { error } = feePayer
+  if (!error) {
+    yield put(setFeePayer({ feePayer }))
+    console.log('the fee payer', yield* select(getFeePayer))
```
Comment (sddioulde) [inline src/store/solana/sagas.ts:15]: going to remove

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-fee-payer

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-fee-payer

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-fee-payer

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-fee-payer

File: src/services/AudiusBackend.js [hunk 3]
```diff
@@ -2438,6 +2440,20 @@ class AudiusBackend {
...
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      }).then(res => res.json())
+    } catch (err) {
+      console.error(err.message)
+      return { error: true }
+    }
+  }
+
```
Comment (piazzatron) [inline src/services/AudiusBackend.js:2456]: Let's have this live in libs - should be a method in the services/identity, and then probably called via SolanaWeb3Manager

File: src/store/solana/sagas.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { put, call, take } from 'typed-redux-saga'
+
+import { setFeePayer } from 'common/store/solana/slice'
+import AudiusBackend from 'services/AudiusBackend'
+import * as backendActions from 'store/backend/actions'
+
+function* watchForFeePayer() {
+  yield take(backendActions.SETUP_BACKEND_SUCCEEDED)
```
Comment (piazzatron) [inline src/store/solana/sagas.ts:8]: can use `waitForBackendSetup`

**Review by piazzatron — APPROVED** [review]
Nice

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-fee-payer

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-fee-payer

**Review by jowlee — APPROVED** [review]

**Pull request merged**